cmd/list.go

```diff
@@ -82,57 +98,141 @@ func list(cmd *cobra.Command) {
 		token = os.Getenv("REANA_ACCESS_TOKEN")
 	}
 	workflow, _ := cmd.Flags().GetString("workflow")
+	listAllSessions, _ := cmd.Flags().GetBool("sessions")
 	jsonOutput, _ := cmd.Flags().GetBool("json")
-	filter, _ := cmd.Flags().GetStringArray("filter")
+	showAll, _ := cmd.Flags().GetBool("all")
+	verbose, _ := cmd.Flags().GetBool("verbose")
+	humanReadable, _ := cmd.Flags().GetBool("human-readable")
+	filters, _ := cmd.Flags().GetStringArray("filter")
+	includeDuration, _ := cmd.Flags().GetBool("include-duration")
+	includeProgress, _ := cmd.Flags().GetBool("include-progress")
+	includeWorkspaceSize, _ := cmd.Flags().GetBool("include-workspace-size")
+	showDeletedRuns, _ := cmd.Flags().GetBool("show-deleted-runs")
+	page, _ := cmd.Flags().GetInt64("page")
+	size, _ := cmd.Flags().GetInt64("size")
 
-	filterNames := []string{"name", "status"}
-	statusFilters, searchFilter := utils.ParseListFilters(filter, filterNames)
+	var runType string
+	if listAllSessions {
+		runType = "interactive"
+	} else {
+		runType = "batch"
+	}
+
+	statusFilters := utils.GetRunStatuses(showDeletedRuns || showAll)
+	var searchFilter string
+	if len(filters) > 0 {
+		filterNames := []string{"name", "status"}
+		statusFilters, searchFilter = utils.ParseListFilters(filters, filterNames)
+	}
 
 	listParams := operations.NewGetWorkflowsParams()
 	listParams.SetAccessToken(&token)
+	listParams.SetType(runType)
+	listParams.SetVerbose(&verbose)
+	listParams.SetPage(&page)
 	listParams.SetWorkflowIDOrName(&workflow)
 	listParams.SetStatus(statusFilters)
 	listParams.SetSearch(&searchFilter)
+	listParams.SetIncludeProgress(&includeProgress)
+	listParams.SetIncludeWorkspaceSize(&includeWorkspaceSize)
+	if size > 0 {
+		listParams.SetSize(&size)
+	}
 
 	listResp, err := client.ApiClient().Operations.GetWorkflows(listParams)
 	if err != nil {
 		fmt.Println("Error: ", err)
 		os.Exit(1)
 	}
+
+	headers := buildListHeaders(
+		runType,
+		verbose,
+		includeWorkspaceSize,
+		includeProgress,
+		includeDuration,
+	)
+	displayListPayload(listResp.Payload, headers[runType], jsonOutput, humanReadable)
+}
+
+func displayListPayload(
+	p *operations.GetWorkflowsOKBody,
+	header []any,
```

Comment:
I don't really like using `any` either, but using `string` has a problem:
- The library used to draw tables had a type `Row`, that's basically a new name to `[]any`
- To not depend in the framework and use it with every type, I made the `utils.DisplayTable` to receive the type `any`
- Unfortunately, GO doesn't let me pass a `string` type when it is expecting `any`, hence why the headers all have `[]any` instead of `[]string`
Do you know a better way to do this?

Reply:
Hmm I see, I think it's worth making the change then